app/controllers/v1/reservations_controller.rb

```diff
@@ -0,0 +1,68 @@
+class V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show edit update destroy]
+
+  # GET /reservations or /reservations.json
+  def index
+    @user = current_user
+    @reservations = Reservation.where(user_id: @user.id)
+    render json: { reservations: @reservations }, status: 200
+  end
+
+  # GET /reservations/1 or /reservations/1.json
+  def show; end
+
+  # GET /reservations/new
+  def new
+    @reservation = Reservation.new
+  end
+
+  # GET /reservations/1/edit
+  def edit
+    @reservation = Reservation.find(params[:id])
+  end
```

Comment:
- [ ] I know your app is not 100% complete yet, but I need to point that you either still need to implement these or remove them if you're not using.